File: libs/Profio.Domain/Entities/Vehicle.cs
using System.Text.Json.Serialization;
using Profio.Domain.Constants;
using Profio.Domain.Interfaces;

namespace Profio.Domain.Entities;

public sealed class Vehicle : IEntity
{
  public string? ZipCodeCurrent { get; set; }
  public string? LicensePlate { get; set; }
  public VehicleType Type { get; set; } = VehicleType.Truck;
  public VehicleStatus Status { get; set; } = VehicleStatus.Idle;
  public string? StaffId { get; set; }

  [JsonIgnore] public Staff? Staff { get; set; }

  public ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();
  public string Id { get; set; } = Ulid.NewUlid().ToString()!;
}

File: apps/profio-app/lib/models/vehicle.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:profio_staff_client/enums/vehicle_status.dart';
import 'package:profio_staff_client/enums/vehicle_type.dart';
import 'package:profio_staff_client/models/staff.dart';

part 'vehicle.g.dart';

@JsonSerializable()
class Vehicle {
  String id;
  String zipCodeCurrent;
  String licensePlate;
  VehicleType type;
  VehicleStatus status;
  Staff? staff;
  Vehicle(
      {this.id = '',
      this.zipCodeCurrent = '',
      this.licensePlate = '',
      this.type = VehicleType.truck,
      this.status = VehicleStatus.idle,
      this.staff});
  factory Vehicle.fromJson(Map<String, dynamic> json) =>
      _$VehicleFromJson(json);

  Map<String, dynamic> toJson() => _$VehicleToJson(this);
}
